### ISSUE TITLE
Client: cannot set no retries or zero retries 

### ISSUE THREAD

**Issue by m-lawliet** [ORESoftware/live-mutex#51]
Hello,

Current implementation doesn't really allow using zero retries.
I saw that "opts.retry" is assigned to a const never used, so it has no effects.
https://github.com/ORESoftware/live-mutex/blob/e5b42d6e954c98ec8e80ee2b07410efe271086b6/src/client.ts#L837

Also, using this kind of assignment:
https://github.com/ORESoftware/live-mutex/blob/e5b42d6e954c98ec8e80ee2b07410efe271086b6/src/client.ts#L300

and this:
https://github.com/ORESoftware/live-mutex/blob/e5b42d6e954c98ec8e80ee2b07410efe271086b6/src/client.ts#L834

results in using 3 as retries value, because "0 || integer > 0" always returns passed integer.
I suggest using the following statements instead, or something similar that suits your stile best:

```
this.lockRetryMax = opts.lockRetryMax || opts.maxRetries || opts.retryMax;
this.lockRetryMax = this.lockRetryMax >= 0 ? this.lockRetryMax : 3;
```
and 

```
const tempRetries = opts.maxRetry || opts.maxRetries;
const maxRetries = tempRetries >= 0 ? tempRetries : this.lockRetryMax;  
```

I am aware that this solution is not as succinct as the original solution, but it allows using zero retries on Client constructor and on acquire calls.
Or you could simply implement behaviour for "opts.retry = false" and avoid doing those changes.

Thanks

**Comment by ORESoftware** [comment]
Got it, let me look into this, will make sure it can work with `0` for `retry` and `retry = false`, give me another day

**Comment by ORESoftware** [comment]
What happens when you do this?

```js
c.lock(<key>, {maxRetries:0}, (e, v) => {});
```

**Comment by ORESoftware** [comment]
I made a fix, and published it here:
0.1.1059-next.0

try installing the exact version above, and it works for you, then I will publish a new patch thanks!

**Comment by m-lawliet** [comment]
Hello, sorry for late response.

You broke something else on `0.1.1059-next.0`, I have some unit tests that got stuck using this version (it seems locks cannot be released, node process remain somehow pending). Using `0.1.1058` this does not happen.

Anyway, I don't really need zero retries anymore although it would be nice to have,
so you should not update this patch right now, since it will probably cause fails in other users too.

Thanks

**Comment by ORESoftware** [comment]
Ok thanks for the heads up! yeah i wrote the patch quickly, so will need to fix that.

**Comment by ORESoftware** [comment]
@m-lawliet I fixed the bug with `0.1.1059-next.0` so live-mutex@0.1.1059 has the no-retry logic you were looking for originally.

thanks!
